tests/components/qbus/test_climate.py

```diff
@@ -0,0 +1,228 @@
+    assert entity
+    assert entity.attributes[ATTR_TEMPERATURE] == _SET_TEMPERATURE
+    assert entity.attributes[ATTR_CURRENT_TEMPERATURE] == _CURRENT_TEMPERATURE
+    assert entity.attributes[ATTR_PRESET_MODE] == _REGIME
+    assert entity.attributes[ATTR_HVAC_ACTION] == HVACAction.IDLE
+    assert entity.state == HVACMode.HEAT
+
+    # After a delay, a full state request should've been sent
+    _wait_and_assert_state_request(hass, mqtt_mock)
+
+    # Simulate a full state response
+    async_fire_mqtt_message(
+        hass, _TOPIC_CLIMATE_STATE, _PAYLOAD_CLIMATE_STATE_PRESET_FULL
+    )
+    await hass.async_block_till_done()
+
+    # Check state after full state response
+    entity = hass.states.get(_CLIMATE_ENTITY_ID)
+    assert entity
+    assert entity.attributes[ATTR_TEMPERATURE] == 22.0
+    assert entity.attributes[ATTR_CURRENT_TEMPERATURE] == _CURRENT_TEMPERATURE
+    assert entity.attributes[ATTR_PRESET_MODE] == _REGIME
+    assert entity.attributes[ATTR_HVAC_ACTION] == HVACAction.HEATING
+    assert entity.state == HVACMode.HEAT
+
+
+async def test_climate_when_invalid_state_received(
+    hass: HomeAssistant,
+    mqtt_mock: MqttMockHAClient,
+    setup_integration: None,
+) -> None:
+    """Test climate when no valid state is received."""
+
+    platform: EntityPlatform = hass.data["entity_components"][CLIMATE_DOMAIN]
+    entity: ClimateEntity = next(
+        (
+            entity
+            for entity in platform.entities
+            if entity.entity_id == _CLIMATE_ENTITY_ID
+        ),
+        None,
+    )
+
+    assert entity
+    entity.async_schedule_update_ha_state = MagicMock()
+
+    # Simulate state response
+    async_fire_mqtt_message(hass, _TOPIC_CLIMATE_STATE, "")
+    await hass.async_block_till_done()
+
+    entity.async_schedule_update_ha_state.assert_not_called()
+
+
+async def test_climate_with_fast_subsequent_changes(
+    hass: HomeAssistant,
+    mqtt_mock: MqttMockHAClient,
+    setup_integration: None,
+) -> None:
+    """Test climate with fast subsequent changes."""
+
+    # Simulate two subsequent partial state responses
+    async_fire_mqtt_message(hass, _TOPIC_CLIMATE_STATE, _PAYLOAD_CLIMATE_STATE_TEMP)
+    await hass.async_block_till_done()
+    async_fire_mqtt_message(hass, _TOPIC_CLIMATE_STATE, _PAYLOAD_CLIMATE_STATE_TEMP)
+    await hass.async_block_till_done()
+
+    # State request should be requested only once
+    _wait_and_assert_state_request(hass, mqtt_mock)
+
+
+async def test_climate_with_unknown_preset(
+    hass: HomeAssistant,
+    mqtt_mock: MqttMockHAClient,
+    setup_integration: None,
+) -> None:
+    """Test climate with passing an unknown preset value."""
+
+    with pytest.raises(ServiceValidationError):
+        await hass.services.async_call(
+            CLIMATE_DOMAIN,
+            SERVICE_SET_PRESET_MODE,
+            {
+                ATTR_ENTITY_ID: _CLIMATE_ENTITY_ID,
+                ATTR_PRESET_MODE: "What is cooler than begin cool?",
```

Comment:
```suggestion
                ATTR_PRESET_MODE: "What is cooler than being cool?",
```
Ice cold! :laughing:

Reply:
Alright alright alright alright alright alright 😄

Reply:
Oh, in all my enthusiasm of you recognizing the song, I forgot to commit the change!